### PR TITLE
Move Intel passes into triton::gpu::intel namespace

### PR DESCRIPTION
**Pull request by etiotto**

### CODE REVIEW

File: third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp [hunk 1]
```diff
@@ -41,7 +37,7 @@ bool isDivisible(Value value, unsigned divisor) {
   if (value.getParentBlock()->isEntryBlock() && isa<BlockArgument>(value)) {
     BlockArgument blockArg = cast<BlockArgument>(value);
     Operation *parentOp = blockArg.getOwner()->getParentOp();
-    if (auto funcOp = dyn_cast<tt::FuncOp>(parentOp)) {
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp:44]: why this change?

File: third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp [hunk 2]
```diff
@@ -320,14 +316,14 @@ struct RewritedInfo {
 // very fragile and to solve we should expose convert Ptr of tensor to a
 // structure contains all values and not only offsets.
 class TritonIntelGPURewriteTensorPointerPass
-    : public impl::TritonIntelGPURewriteTensorPointerBase<
+    : public triton::gpu::intel::impl::TritonIntelGPURewriteTensorPointerBase<
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp:319]:
```suggestion
    : public ttgi::impl::TritonIntelGPURewriteTensorPointerBase<
```

File: third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp [hunk 3]
```diff
@@ -320,14 +316,14 @@ struct RewritedInfo {
...
           TritonIntelGPURewriteTensorPointerPass> {
 private:
   DenseMap<Value, RewritedInfo> rewritedInfo;
   DenseSet<Value> valueToRemove;
 
 public:
-  using impl::TritonIntelGPURewriteTensorPointerBase<
+  using triton::gpu::intel::impl::TritonIntelGPURewriteTensorPointerBase<
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp:326]:
```suggestion
  using ttgi::impl::TritonIntelGPURewriteTensorPointerBase<
```

File: third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp [hunk 4]
```diff
@@ -23,16 +11,24 @@ namespace ttg = mlir::triton::gpu;
 namespace ttgi = mlir::triton::gpu::intel;
 
 namespace mlir {
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp:13]:
```suggestion
namespace mlir::triton::gpu::intel {
```

File: third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp:1]: similar comments for other files.

File: third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp (hunk 4 above)
Comment (etiotto) [reply third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp:13]: done

Comment (etiotto) [reply third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp:13]: done

File: third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp (hunk 1 above)
Comment (etiotto) [reply third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp:44]: reverted

File: third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp (hunk 3 above)
Comment (etiotto) [reply third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp:326]: ditto as above

File: third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp (hunk 2 above)
Comment (etiotto) [reply third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp:319]: I prefer the way I wrote it because is explicit. In general I think is better to limit using declarations (at the file scope level) like `tt`, `ttg`, `ttgi` ... as they just hide the namespace and require the reader to look the using definition up.

File: third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp
Comment (etiotto) [reply third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp:1]: done

File: third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp (hunk 2 above)
Comment (whitneywhtsang) [reply third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp:311]: was thinking to use it as `ttgi` is defined.

**Review by whitneywhtsang — APPROVED** [review]
